### ISSUE TITLE
Quality Control, Testing, Improving, and Development Phases

### ISSUE THREAD

**Issue by Anonymous-Koala** [ProbablyTez/Agony#470]
So, to make sure everything is up to quality I've started thinking of ways to ensure quality control. This is what I've came up with so far:

**Room Design**
-Test a room with the following, each in their own time:
Low Speed(So it's not unfair damage), Low Firerate, Low Damage, Flight(So it's not too easy), Base Isaac without any items. Test every scenario a few times to make sure it can consistently be done without being hit.

-No "meme rooms", which means no rooms with golden poops spelling Isaac for example, rooms with free things with no catch/thing blocking you from them.

**Tear Effects**
-In addition to the obvious item guidelines that you can find [here](https://docs.google.com/document/d/1Oz3oQJb9SHk2f-BixH-AIAJTGR2JSsD5odTS2c7k8Tk/edit) , Tear Effects require synergy checks.
-Make a list of all items who modify tears, and the ones agony adds who do that, and start thinking about synergies for them. Tear Modifiers should synergize with as much as they can, ESPECIALLY the "Synergy Killers"(Brimstone, Mom's Knife, Dr. and Epic Fetus).

**Enemies and Bosses**
-Designed according to [this](https://docs.google.com/document/d/1QsOByKQCfIBUarnRtXppvIKFvZHz4Z4bjofaAW6Xx-Q/edit) .
-Must be tested several times in each room they're in, according to the same terms that the rooms have to be tested, except for flight, and with the added test, "can you beat it without taking damage with 2X random items, X being the floor it spawns on (Basement = 1 and so on)".

And now for the other thing, the development phases.

**Phase 1**
-Making ideas and suggestions, building up the framework for the mod, as well as spriting, coding, and implementing ideas as we go along.
-During this phase we'll also do a lot of approvals and denials on the discord on a semi regular basis. 

**Phase 2**
-During phase 2 we'll mostly be doing these things. This phase starts once we have enough ideas to truly focus on the implementation of them and testing/quality control. The things we must do are the content for the existing levels, and the alts.

**Phase 3**
-Purely quality control and tweaking. This is the final phase essentially. We'll talk more about this one when we're close to it, which isn't now. 

**Currently we're in:** Phase 1.

**Comment by ProbablyTez** [comment]
Every room shouldnt be beatable with base items, otherwise we'll end up with super easy room in the late game
Meme rooms are okay if *extremely* rare. Like one in 100 runs is a good frequency

Other than that I agree with these guidelines

**Comment by Anonymous-Koala** [comment]
Fair enough about the first point. The latter depends. Generally, try to stay above ab+ rooms, imo.

**Comment by Anonymous-Koala** [comment]
Also, albeit a bit late of a response, Meme Rooms are bad as they give nothing of value. Their only purpose is for someone to screenshot them to reddit and go all "xd upboats". Everyone prefer having actual good rooms rather than that.

**Comment by dedChar** [comment]
What would a meme room even contain

**Comment by CrimsonCube** [comment]
golden poops spelling Isaac

**Comment by CrimsonCube** [comment]
Jokes aside though, I personally don't thing there should be no rooms with for example "free things with no catch/thing blocking you from them". In my opinion it's nice to have some very rare but very good things happen, so that every once in a while a player can experience a run where game is more generous that usual. Especially since Isaac can be pretty unforgiving sometimes, so it would be a nice counterweight. Just as Tez said, "like one in 100 runs".

**Comment by Anonymous-Koala** [comment]
Thing is it has to be done in a way not like AB+. AB+ rooms in general are bad tbh. Either meme rooms(everything focusing around greed just being shoved into one room for example), or way too generous for no reason.
